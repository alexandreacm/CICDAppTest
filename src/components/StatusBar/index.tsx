import React from 'react';
import { StatusBar as RNStatusBar } from 'react-native';

type Props = {
  isDarkMode: boolean;
}

const StatusBar: React.FC<Props> = ({ isDarkMode }) => {
  return (
    <RNStatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
  );
};

export default StatusBar;
