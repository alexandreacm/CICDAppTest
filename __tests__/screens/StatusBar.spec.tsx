import React from 'react';
import { render } from '@testing-library/react-native';
import StatusBar from '../../src/components/StatusBar';

describe('<StatusBar />', () => {
  beforeEach(() => {
    jest.resetModules();
  });
  test('Should renders correctly',  () => {
    render(<StatusBar isDarkMode={false} />);
  });

  test('Should renders snapshot',  () => {
    const { toJSON } = render(<StatusBar isDarkMode={false} />);

    expect(toJSON()).toMatchSnapshot();
  });



  test('Should renders correctly with dark mode',  () => {
    render(<StatusBar isDarkMode={true} />);
  });
});
