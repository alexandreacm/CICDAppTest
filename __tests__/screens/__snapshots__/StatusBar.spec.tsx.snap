// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBar /> Should renders snapshot 1`] = `null`;
